src/sentry/scim/endpoints/users.py

```diff
@@ -0,0 +1,154 @@
+
+    def get(self, request, organization, member):
+        context = serialize(member, serializer=OrganizationMemberSCIMSerializer())
+        return Response(context)
+
+    def patch(self, request, organization, member):
+        for operation in request.data.get("Operations", []):
+            # we only support setting active to False which deletes the orgmember
+            if operation["value"]["active"] is False:
+                self._delete_member(request, organization, member)
+                return Response(status=204)
+        context = serialize(member, serializer=OrganizationMemberSCIMSerializer())
+        return Response(context)
+
+    def delete(self, request, organization, member):
+        self._delete_member(request, organization, member)
+        return Response(status=204)
+
+
+class OrganizationSCIMUserIndex(SCIMEndpoint):
+    def get(self, request, organization):
+        # note that SCIM doesn't care about changing results as they're queried
+        # TODO: sanitize get parameter inputs?
+        try:
+            filter_val = parse_filter_conditions(request.GET.get("filter"))
+        except Exception:
+            return Response(detail=SCIM_400_INVALID_FILTER, status=400)
+
+        queryset = (
+            OrganizationMember.objects.filter(
+                Q(invite_status=InviteStatus.APPROVED.value),
+                Q(user__is_active=True) | Q(user__isnull=True),
+                organization=organization,
+            )
+            .select_related("user")
+            .order_by("email", "user__email")
+        )
+        if filter_val:
+            queryset = queryset.filter(
+                Q(email__in=filter_val) | Q(user__email__in=filter_val)
+            )  # not including secondary email vals (dups, etc.)
+
+        # TODO: how is queryset ordered?
+
+        def data_fn(offset, limit):
+            return list(queryset[offset : offset + limit])
+
+        return self.paginate(
+            request=request,
+            on_results=lambda results: serialize(results, None, OrganizationMemberSCIMSerializer()),
+            paginator=GenericOffsetPaginator(data_fn=data_fn),
+            default_per_page=int(request.GET.get("count", 100)),
+            queryset=queryset,
+        )
+
+    def post(self, request, organization):
+        # TODO: confirm mixed case emails get converted to lowercase
+        serializer = OrganizationMemberSerializer(
+            data={
+                "email": request.data.get("userName"),
+                "role": roles.get(organization.default_role).id,
+            },
+            context={
+                "organization": organization,
+                "allowed_roles": [roles.get(organization.default_role)],
+                "allow_existing_invite_request": True,
+            },
+        )
+
+        if not serializer.is_valid():
+            if (
+                "email" in serializer.errors
+                and "is already a member" in serializer.errors["email"][0]
```

Comment:
Yikes. Looks like we're stuck with scanning the string because `BaseSerializer` (from Python's `rest_framework` module) makes the original `ValidationError` objects inaccessible?

Is there a way around it by calling `run_validation` instead of `is_valid` and catching the error ourselves? If so, we could go to the source of the original error and have it raise a special subclass of `ValidationError` that we can check for here.

Also, I'd recommend getting rid of the `[0]` by
```python
and any(("is already a member" in error) for error in serializer.errors["email"])
```

Reply:
unfortunately no, looking at the internals whatever errors are raised they are always clobbered by serialize function into a ValidationError. could probably do something where we override a serializer method, but the string parsing hack is simpler. 

we could also refactor and remove the conflict logic entirely from the serializer. funny to see this comment haha https://github.com/getsentry/sentry/pull/15027#discussion_r333282009

Reply:
> we could also refactor and remove the conflict logic entirely from the serializer.

I'd support this long-term. I coincidentally ran into a similar problem with the org requirement settings recently. I think some of our serializers Do Too Much when we rely on them to validate the integrity of the application state, as opposed to only the input/output they're touching.

I'm fine with this as-is for the current PR, however.